flask/app/routes.py

```diff
@@ -179,33 +179,45 @@ def mixin(entity: db.Model, json_mixin: Dict[str, Any], columns: List[str]) -> U
             setattr(entity, field, value)
 
 
-@app.route('/api/participants/<int:participant_id>', methods = ['PATCH'])
+@app.route('/api/<model_name>/<int:id>', methods = ['PATCH'])
 @login_required
-def update_participants(participant_id: int):
+def update_entity(model_name:str, id:int):
     if not request.json:
         return 'Request body must be JSON', 415
 
-    participant = models.Participant.query.get(participant_id)
-    if not participant:
```

Comment:
the `None` check got deleted but it's necessary because we want to 404 if the entity does not exist, not 500 trying to access a property of `None`